src/oss-tests/Helpers/PackageActionsHelper.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Tests.Helpers;
+
+using Contracts;
+using Model.Metadata;
+using Moq;
+using NuGet.Versioning;
+using PackageUrl;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+public static class PackageActionsHelper<T> where T : IManagerPackageVersionMetadata
+{
+    /// <summary>
+    /// Set up a mock of <see cref="IManagerPackageActions{T}"/> for this test run.
+    /// </summary>
+    /// <param name="purl">The <see cref="PackageURL"/> to use when configuring the mocked calls for this manager.</param>
+    /// <param name="metadata">The <see cref="T"/> to use when returning the call to <see cref="IManagerPackageActions{T}.GetMetadataAsync"/>.</param>
+    /// <param name="versions">The list of versions to return when mocking the call 
+    /// to <see cref="IManagerPackageActions{T}.GetAllVersionsAsync"/>.</param>
+    /// <param name="validSquats">The list of squats to populate the mock to <see cref="IManagerPackageActions{T}.DoesPackageExistAsync"/>.</param>
+    /// <param name="includePrerelease">If pre-release/beta versions should be included.</param>
+    /// <returns>A Mocked <see cref="IManagerPackageActions{T}"/>.</returns>
+    public static IManagerPackageActions<T>? SetupPackageActions(
+        PackageURL? purl = null,
+        T? metadata = default,
+        IEnumerable<string>? versions = null,
+        IEnumerable<string>? validSquats = null,
+        bool includePrerelease = true) 
+    {
+        Mock<IManagerPackageActions<T>> mockPackageActions = new();
+
+        if (purl is not null)
+        {
+            if (metadata is not null)
+            {
+                // Mock the metadata call if metadata was provided.
+                mockPackageActions.Setup(actions => actions.GetMetadataAsync(
+                    It.Is<PackageURL>(p => p.Name.Equals(purl.Name) && (purl.Version == null || p.Version.Equals(purl.Version))), It.IsAny<bool>(), It.IsAny<CancellationToken>()).Result).Returns(
+                    metadata);
+            }
+
+            if (versions is not null)
+            {
+                // Mock the list of versions if the list was provided.
+                IEnumerable<string> versionsArray = versions as string[] ?? versions.ToArray();
+                mockPackageActions.Setup(actions => actions.GetAllVersionsAsync(
+                    It.Is<PackageURL>(p => p.Name.Equals(purl.Name)), 
+                    includePrerelease, 
+                    It.IsAny<bool>(), 
+                    It.IsAny<CancellationToken>()).Result)
+                    .Returns(() =>
+                    {
+                        return typeof(T) == typeof(NuGetPackageVersionMetadata)
+                            ? versionsArray.Where(v => includePrerelease || !NuGetVersion.Parse(v).IsPrerelease)
+                            : versionsArray;
```

Comment:
Nit: this seems a bit odd. Would you mind elaborating on why this special case filtering is here while setting up mocks? Is this because we load in sample metadata from a file or something?

Reply:
Yeah, we load it in from the direct API response as json, I just reimplement the includePrerelease logic check here as well.